pex/environment.py

```diff
@@ -120,13 +120,14 @@ def __init__(self, pex, pex_info, interpreter=None, **kw):
     self._interpreter = interpreter or PythonInterpreter.get()
     self._inherit_path = pex_info.inherit_path
     self._supported_tags = []
+
+    platform = Platform.current()
     super(PEXEnvironment, self).__init__(
       search_path=[] if pex_info.inherit_path == 'false' else sys.path,
+      platform=platform.platform,
```

Comment:
`platform.platform` is a bit confusing 🤔. I think it's outside the scope of this PR to rename the `Platform` object and/or its field `.platform`, but in the meantime this variable could be renamed perhaps, or maybe a brief note what the difference is.

Reply:
Explicitified